src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Pattern.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertIfToSwitch
+{
+    partial class CSharpConvertIfToSwitchCodeRefactoringProvider
+    {
+        private abstract class Pattern : IPattern
+        {
+            SyntaxNode IPattern.CreateSwitchLabel() => CreateSwitchLabel();
+
+            protected abstract CasePatternSwitchLabelSyntax CreateSwitchLabel();
```

Comment:
This is where contravariant overrides could help (only if they work with implicit interface implementations), anyways, a generic `Pattern<T>` could be also helpful. I'll make it so.